$NetBSD: patch-ak,v 1.1 2008/07/24 01:07:31 bjs Exp $

ConnDis: properly cast 'addr' before accessing it as a byte array.

Bug #15884: Remove useless sleep()'s from the connection code.

--- src/ConnDis.c.orig	2008-03-06 15:45:06.000000000 -0500
+++ src/ConnDis.c
@@ -42,6 +42,7 @@ in this Software without prior written a
 #include <X11/Xdmcp.h>
 #include <stdio.h>
 #include <ctype.h>
+#include <unistd.h>
 
 #if !defined(WIN32)
 #ifndef Lynx
@@ -389,10 +390,7 @@ _X11TransConnectDisplay (
 	    trans_conn = NULL;
 
 	    if (connect_stat == TRANS_TRY_CONNECT_AGAIN)
-	    {
-		sleep(1);
 		continue;
-	    }
 	    else
 		break;
 	    }
@@ -409,7 +407,6 @@ _X11TransConnectDisplay (
 	    {
 	    _X11TransClose(trans_conn);
 	    trans_conn = NULL;
-	    sleep(1);
 	    if (saddr)
 	    {
 		free ((char *) saddr);
@@ -1143,7 +1140,7 @@ GetAuthorization(
 	    
 	    /* In the case of v4 mapped addresses send the v4 
 	       part of the address - addr is already in network byte order */
-	    if (memcmp(addr+8, ipv4mappedprefix, 12) == 0) {
+	    if (memcmp((char*)addr+8, ipv4mappedprefix, 12) == 0) {
 		for (i = 20 ; i < 24; i++)
 		    xdmcp_data[j++] = ((char *)addr)[i];
 	    
